components/camel-observation/src/test/java/org/apache/camel/observation/CamelMicrometerObservationTestSupport.java

```diff
@@ -0,0 +1,255 @@
+
+    private MeterRegistry meterRegistry = new SimpleMeterRegistry();
+
+    private ObservationRegistry observationRegistry;
+
+    CamelMicrometerObservationTestSupport(SpanTestData[] expected) {
+        this.expected = expected;
+    }
+
+    @Override
+    protected CamelContext createCamelContext() throws Exception {
+        CamelContext context = super.createCamelContext();
+        micrometerObservationTracer = new MicrometerObservationTracer();
+
+        tracerFactory = SdkTracerProvider.builder()
+                .addSpanProcessor(new LoggingSpanProcessor())
+                .addSpanProcessor(SimpleSpanProcessor.create(inMemorySpanExporter)).build();
+
+        tracer = tracerFactory.get("tracerTest");
+
+        observationRegistry = ObservationRegistry.create();
+        observationRegistry.observationConfig().observationHandler(new DefaultMeterObservationHandler(meterRegistry));
+
+        io.micrometer.tracing.Tracer otelTracer = otelTracer();
+        OtelPropagator otelPropagator = new OtelPropagator(ContextPropagators.create(B3Propagator.injectingSingleHeader()), tracer);
+        observationRegistry.observationConfig().observationHandler(new ObservationHandler.FirstMatchingCompositeObservationHandler(new CamelPropagatingSenderTracingObservationHandler<>(otelTracer, otelPropagator), new CamelPropagatingReceiverTracingObservationHandler<>(otelTracer, otelPropagator), new CamelDefaultTracingObservationHandler(otelTracer)));
+
+        micrometerObservationTracer.setObservationRegistry(observationRegistry);
+        // if you want baggage
+        micrometerObservationTracer.setTracer(otelTracer);
+        micrometerObservationTracer.setExcludePatterns(getExcludePatterns());
+        micrometerObservationTracer.addDecorator(new TestSEDASpanDecorator());
+        micrometerObservationTracer.init(context);
+        return context;
+    }
+
+    private OtelTracer otelTracer() {
+        OtelCurrentTraceContext otelCurrentTraceContext = new OtelCurrentTraceContext();
+        OtelBaggageManager otelBaggageManager = new OtelBaggageManager(otelCurrentTraceContext, Collections.emptyList(), Collections.emptyList());
+        return new OtelTracer(tracer, otelCurrentTraceContext, o -> {
+
+        }, otelBaggageManager);
+    }
+
+    protected Set<String> getExcludePatterns() {
+        return new HashSet<>();
+    }
+
+    protected void verify() {
+        verify(expected, false);
+    }
+
+    protected void verify(boolean async) {
+        verify(expected, async);
+    }
+
+    protected List<SpanData> verify(SpanTestData[] expected, boolean async) {
+        List<SpanData> spans = inMemorySpanExporter.getFinishedSpanItems();
+        spans.forEach(mockSpan -> {
+            System.out.println("Span: " + mockSpan);
+            System.out.println("\tComponent: " + mockSpan.getAttributes().get(COMPONENT_KEY));
+            System.out.println("\tTags: " + mockSpan.getAttributes());
+            System.out.println("\tLogs: ");
```

Comment:
This should go to a log file, to make it easier for us in the CI.

Reply:
Just FYI I've literally copied that from OpenTelemetry code.

Reply:
Thanks for the heads up! I will fix it there.